pkg/util/iptables/iptables.go

```diff
@@ -783,3 +789,90 @@ func isResourceError(err error) bool {
 	}
 	return false
 }
+
+// ParseError records the payload when iptables restore fails
+type ParseError interface {
+	// Line returns the number of lines on which iptables-restore is failing
+	// NOTE: First line is line 1
+	Line() int
+	// Error returns the error message of the location where the iptables-restore execution failed
+	Error() string
+}
+
+type parseError struct {
+	cmd  string
+	line int
+}
+
+func (e parseError) Line() int {
+	return e.line
+}
+
+func (e parseError) Error() string {
+	return fmt.Sprintf("%s: input error on line %d: ", e.cmd, e.line)
+}
+
+// LineData is used to save the line of the lines and its corresponding data
+type LineData struct {
+	// Number of rows in lines, First line is line 1
+	Line int
+	// The date of the line
+	Data string
+}
+
+var regexpParseError = regexp.MustCompile("line ([1-9][0-9]*) failed$")
+
+// parseRestoreError extracts the line from the error, if it matches returns parseError
+// for example:
+// input: iptables-restore: line 51 failed
+// output: parseError:  cmd = iptables-restore, line = 51
+// NOTE: parseRestoreError depends on the error format of iptables, if it ever changes
+// we need to update this function
+func parseRestoreError(err string) (ParseError, bool) {
```

Comment:
by convention `err` is type `error`.  Can you rename the argument here to "str" or something, and then use `err` instead of `errMsg` below?

Reply:
done